libraries/scripting/dependencies-maven-all/build.gradle.kts

```diff
@@ -0,0 +1,89 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+description = "Shaded Maven dependencies resolver"
+
+val JDK_18: String by rootProject.extra
+val jarBaseName = property("archivesBaseName") as String
+
+val embedded by configurations
+
+embedded.apply {
+    exclude("org.slf4j", "slf4j-api")
+    exclude("org.eclipse.aether", "aether-api")
+    exclude("org.eclipse.aether", "aether-util")
+    exclude("org.eclipse.aether", "aether-spi")
+}
+
+plugins {
+    kotlin("jvm")
+    id("jps-compatible")
+}
+
+dependencies {
+    embedded(project(":kotlin-scripting-dependencies-maven")) { isTransitive = false }
+    embedded(project(":kotlin-scripting-dependencies")) { isTransitive = false }
+
+    embedded("org.eclipse.aether:aether-connector-basic:1.1.0")
+    embedded("org.eclipse.aether:aether-transport-wagon:1.1.0")
+    embedded("org.eclipse.aether:aether-transport-file:1.1.0")
+    embedded("org.apache.maven:maven-core:3.8.1")
+    embedded("org.apache.maven.wagon:wagon-http:3.4.3")
+}
+
+sourceSets {
+    "main" {}
+    "test" {}
+}
+
+publish()
+
+noDefaultJar()
+sourcesJar()
+javadocJar()
+
+val relocatedJar by task<ShadowJar> {
+    configurations = listOf(embedded)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
+    destinationDirectory.set(File(buildDir, "libs"))
+    archiveClassifier.set("before-proguard")
+
+    transform(ComponentsXmlResourceTransformerPatched())
+
+    if (kotlinBuildProperties.relocation) {
+        packagesToRelocate.forEach {
+            relocate(it, "$kotlinEmbeddableRootPackage.$it")
+        }
+    }
+}
+
+val proguard by task<CacheableProguardTask> {
```

Comment:
please synchronize with the latest changes in the main-kts build file.

Reply:
Done